app/portal/courses/[id]/page.js

```diff
@@ -98,14 +75,111 @@ export default function Page() {
   // Selected Course Data
   const [selectedCourse, setSelectedCourse] = useState(null)
   const [selectedMaterials, setSelectedMaterials] = useState(null)
+  const [formState, setFormState] = useState(initialState)
   const handleOpenMaterialNew = () => setOpenMaterialNew(true)
   const handleCloseMaterialNew = () => setOpenMaterialNew(false)
   const { courses, course_materials } = useData();
+  const axiosInstance = AxiosWithAuth()
 
   const pathname = usePathname()
   const regex = /-/g
   const newStr = pathname.slice(16).replace(regex, " ")
 
+  const getMaterialType = (material) => {
```

Comment:
We can leave this here, but the initial intent was for the material types to be created and managed from within the admin page. This will work from now, and maybe suffice for the project itself? We can consider that later.

Reply:
Right! To clarify, this piece simply solves the matter of a user adding a new course material, the material getting added to the data grid, and the "Material Type" not rendering.

When the database sends back a positive response from adding a new material, it doesn't include the `material_type` as a string - instead it returns a `material_type_id`. This function simply reads that ID and returns the appropriate string for displaying the new course to the data grid without requiring a full refresh. :)